### PR TITLE
Fix HTTP headers for issue attachment download

### PR DESCRIPTION
**Pull request by andreynering**
This fixes:

- Download filename was wrong for files other than images. Example: It was `download` instead of `file.pdf`
- PDF was downloading instead of showing on browser

### CODE REVIEW

File: cmd/web.go [hunk 1]
```diff
@@ -336,11 +336,7 @@ func runWeb(ctx *cli.Context) error {
 			}
 			defer fr.Close()
 
-			ctx.Header().Set("Cache-Control", "public,max-age=86400")
-			ctx.Header().Set("Content-Disposition", fmt.Sprintf(`inline; filename="%s"`, attach.Name))
-			// Fix #312. Attachments with , in their name are not handled correctly by Google Chrome.
-			// We must put the name in " manually.
-			if err = repo.ServeData(ctx, "\""+attach.Name+"\"", fr); err != nil {
+			if err = repo.ServeData(ctx, attach.Name, fr); err != nil {
```
Comment (lunny) [inline cmd/web.go]: why removed `"`?

Comment (strk) [inline cmd/web.go:339]: Does this re-introduce the referenced issue ?
https://github.com/gogits/gogs/issues/312

File: routers/repo/download.go [hunk 2]
```diff
@@ -22,14 +22,16 @@ func ServeData(ctx *context.Context, name string, reader io.Reader) error {
 		buf = buf[:n]
 	}
 
-	if !base.IsTextFile(buf) {
-		if !base.IsImageFile(buf) {
-			ctx.Resp.Header().Set("Content-Disposition", "attachment; filename=\""+path.Base(ctx.Repo.TreePath)+"\"")
-			ctx.Resp.Header().Set("Content-Transfer-Encoding", "binary")
-		}
-	} else if !ctx.QueryBool("render") {
+	ctx.Resp.Header().Set("Cache-Control", "public,max-age=86400")
```
Comment (strk) [inline routers/repo/download.go:26]: This cache lifetime change seems to be unrelated to the scope of this PR

File: cmd/web.go (hunk 1 above)
Comment (andreynering) [reply cmd/web.go:339]: No, the quote is preserved [here][here1] and [here][here2]

[here1]: https://github.com/go-gitea/gitea/pull/270/files#diff-c333e0fd3bba06cfaf6456df14c8e32dR30
[here2]: https://github.com/go-gitea/gitea/pull/270/files#diff-c333e0fd3bba06cfaf6456df14c8e32dR32

File: routers/repo/download.go (hunk 2 above)
Comment (andreynering) [reply routers/repo/download.go:26]: The cache was there before, I just moved it to another place

Comment (strk) [reply routers/repo/download.go:26]: Ah, now I see, thanks for double-checking

File: cmd/web.go (hunk 1 above)
Comment (strk) [reply cmd/web.go:339]: Did you _test_ the issue explained in gogits#312 ?
I'm not sure the bug was related to the quotes in Content-Disposition rather than in repo.ServeData argument

Comment (andreynering) [reply cmd/web.go:339]: Hmm... In fact there's a problem, thanks for checking.

But I don't understand why it happen. I just moved the quotes, but it's there.

Comment (andreynering) [reply cmd/web.go:339]: @strk It's fixed now. It's was not actually fixed before, because the header was not actually set for these cases.

File: routers/repo/download.go [hunk 3]
```diff
@@ -22,14 +23,19 @@ func ServeData(ctx *context.Context, name string, reader io.Reader) error {
...
+
+	// Google Chrome dislike commas in filenames, so let's change it to a space
+	name = strings.Replace(name, ",", " ", -1)
```
Comment (strk) [inline routers/repo/download.go:29]: *I* dislike spaces in filenames, how about a dash ? :)
Actually, how about a sanitizing function to generalize the issue ? Like, I'm thinking slashes might be impractical to have...

Comment (vtolstov) [reply routers/repo/download.go:29]: i like underscores....

**Comment by strk** [comment]
underscores work for me too, although I prefer dashes as you can
do that without holding SHIFT :)

File: routers/repo/download.go (hunk 3 above)
Comment (andreynering) [reply routers/repo/download.go:29]: > Actually, how about a sanitizing function to generalize the issue ? Like, I'm thinking slashes might be impractical to have...

I did research on this and tried few approaches. Unfortunately there isn't how to sanitize it, it's actually a Chrome bug. And only commas trigger this AFAIK.

I don't have strong feeling for comma vs space.

**Comment by strk** [comment]
LGTM - the systems I use don't choke on spaces (it's just me, but I can blame the people who attach files with spaces or commas in their filenames)

**Comment by thibaultmeyer** [comment]
LGTM
